sources/kafka/__init__.py

```diff
@@ -0,0 +1,96 @@
+"""A source to extract Kafka messages.
+
+When extraction starts, partitions length is checked -
+data is read only up to it, overriding the default Kafka's
+behavior of waiting for new messages in endless loop.
+"""
+from contextlib import closing
+from typing import Any, Callable, Dict, Iterable, List, Optional, Union
+
+from confluent_kafka import Consumer, Message  # type: ignore
+
+import dlt
+from dlt.common import logger
+from dlt.common.time import ensure_pendulum_datetime
+from dlt.common.typing import TDataItem, TAnyDateTime
+from .helpers import (
+    default_msg_processor,
+    KafkaCredentials,
+    OffsetTracker,
+)
+
+
+@dlt.resource(
+    name="kafka_messages",
+    table_name=lambda msg: msg["_kafka"]["topic"],
+    standalone=True,
+)
+def kafka_consumer(
+    topics: Union[str, List[str]],
+    credentials: Union[KafkaCredentials, Consumer] = dlt.secrets.value,
+    msg_processor: Optional[
+        Callable[[Message], Dict[str, Any]]
+    ] = default_msg_processor,
+    batch_size: Optional[int] = 3000,
```

Comment:
should this also be configurable?

Reply:
good catch, it should be configurable, as the final user might want to have an initial batch size big enough to cover the all topic load.

Reply:
Well, that's a source argument. It's configurable, `3000` is just a default value.

Reply:
@sultaniman @nicor88 do you mean that `batch_size` should be configurable via a [Config value](https://dlthub.com/docs/general-usage/credentials/configuration)?

Reply:
I realised that `batch_size ` is indeed a configurable - therefore this point is addressed already.